connectors/sources/google_cloud_storage.py

```diff
@@ -199,10 +201,50 @@ async def _api_call(
                 )
                 await asyncio.sleep(DEFAULT_WAIT_MULTIPLIER**retry_counter)
 
+    def get_pem_format(self, private_key, max_split=-1):
+        """Convert key into PEM format.
+
+        Args:
+            private_key (str): Private_key in raw format.
+            max_split (int): Specifies how many splits to do. Defaults to -1.
+
+        Returns:
+            string: PEM format
+        """
+        private_key = private_key.replace(" ", "\n")
+        private_key = " ".join(private_key.split("\n", max_split))
+        private_key = " ".join(private_key.rsplit("\n", max_split))
+        return private_key
+
+    def _initialize_configurations(self):
+        """Initialize the ServiceAccountCreds"""
+        if self.service_account_credentials is not None:
+            return
+
+        json_credentials = json.loads(self.configuration["service_account_credentials"])
+
+        if (
+            json_credentials.get("private_key")
+            and "\n" not in json_credentials["private_key"]
```

Comment:
Just curious what's the purpose of `\n` check.

What if a user accidentally adds a linebreak in Kibana?

Reply:
> What if a user accidentally adds a linebreak in Kibana?

The service account credential generated from the Google Cloud includes `\n` for line breaks [[ref](https://cloud.google.com/iam/docs/keys-create-delete#creating)], which is expected by the package. but, when there will be `" "`(line breaks-incase if it's modified by the user) then only we will have to replace it with '\n'.

Reply:
In `get_pem_format`, it seems you try to first replace `" "` with `"\n"`, then replace `"\n"` with `" "` again?

What's the purpose of `get_pem_format`? If the credential is modified by the user, should we just raise an error?

I see it just tried to escape unicode originally:
https://github.com/elastic/connectors-python/blob/4ec45095b65e1ba2c2dd64496a67ad040d029e03/connectors/sources/google_cloud_storage.py#L78-L83

Reply:
`get_pem_format` adds new line inplace of `" "`. for reference I've added a [test](https://github.com/elastic/connectors-python/pull/493/files#diff-ebc6238e4238c85b506e32488d7fceccd540da7ffbf05065475f0379721545b6R539) to illustrate the same.

> What's the purpose of `get_pem_format`? If the credential is modified by the user, should we just raise an error?

The package itself will raise an SSL error if the key is malformed. But, I added support for the `get_pem_format` keeping this [Google cloud storage add public key](https://cloud.google.com/iam/docs/keys-upload#uploading) in mind, which enables users to add personal Public-Private Key pair and use those to authenticate with the Google Cloud. In this case, user might add Private-Key with `" "` in it instead of `\n`.